src/get-started/create-service.md

```diff
@@ -371,72 +362,109 @@ executed:
-    }
-}
 ```
 
+Deriving the `ExecutionFail` trait here will make our errors generic and
+compatible with other error kinds used within Exonum blockchain, this macro
+is similar to the `failure::Fail`.
+
 ### Transaction Execution
 
-Every transaction in Exonum has business logic of the blockchain attached,
-which is encapsulated in the `Transaction` trait.
-This trait has `execute` method which contains logic applied to the
-storage when a transaction is executed.
+Above we've defined the interface of our service, but currently this interface
+has no implementation. Thus, there is no actual business logic attached to
+them. To fix this situation, we should declare our service, and then implement
+the `CryptocurrencyInterface` trait for it.
+
+The service declaration is similar to the usual structure declaration:
 
-In our case `execute` method gets the reference to the `TransactionContext`.
-It includes `Fork` of the storage (can be accessed with `.fork()`) and the
-public key which was used to sign the transaction (can be accessed with
-`.author()`). We wrap `Fork` with our `CurrencySchema` to access our data
-layout.
+```rust
+/// Cryptocurrency service implementation.
+#[derive(Debug, ServiceFactory, ServiceDispatcher)]
+#[service_dispatcher(implements("CryptocurrencyInterface"))]
+#[service_factory(proto_sources = "crate::proto")]
+pub struct CryptocurrencyService;
+
+impl Service for CryptocurrencyService {}
+```
+
+`service_dispatcher` macro collects information about interfaces implemented
+by service, and `service_factory` macro generates a code to create instances
+of our service, similarly to the [Factory method design pattern][factory-method].
+
+[factory-method]: https://en.wikipedia.org/wiki/Factory_method_pattern
+
+The implementation of the `Service` trait can contain the additional elements of
+the service lifecycle, like wiring the API. Currently we can skip them and
+leave the implementation empty.
+
+Now, when we have the structure, we can implement the actual business logic.
+
+We will do it in two steps, one step for each transaction we have.
 
 For creating a wallet, we check that the wallet does not exist and add a new
 wallet if so:
 
 ```rust
-impl Transaction for TxCreateWallet {
-    fn execute(&self, mut context: TransactionContext) -> ExecutionResult {
-        let author = context.author();
-        let view = context.fork();
-        let mut schema = CurrencySchema::new(view);
-        if schema.wallet(&author).is_none() {
-            let wallet = Wallet::new(&author, &self.name, INIT_BALANCE);
+impl CryptocurrencyInterface<ExecutionContext<'_>> for CryptocurrencyService {
+    type Output = Result<(), ExecutionError>;
+
+    fn create_wallet(
+        &self,
+        context: ExecutionContext<'_>,
+        arg: TxCreateWallet
+    ) -> Self::Output {
+        let author = context
+            .caller()
+            .author()
+            .expect("Wrong 'TxCreateWallet' initiator");
```

Comment:
I wonder if's worth explaining that extracting `author` like this is not the best idea in production (`expect` should not be used; `CallerAddress` should be used for better forward compatibility).